p2p/src/main/kotlin/maru/p2p/P2PManager.kt

```diff
@@ -0,0 +1,130 @@
+/*
+   Copyright 2025 Consensys Software Inc.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package maru.p2p
+
+import io.libp2p.core.crypto.unmarshalPrivateKey
+import java.nio.file.Path
+import java.util.Optional
+import java.util.concurrent.TimeUnit
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+import tech.pegasys.teku.infrastructure.async.SafeFuture
+import tech.pegasys.teku.networking.p2p.libp2p.MultiaddrPeerAddress
+import tech.pegasys.teku.networking.p2p.libp2p.PeerAlreadyConnectedException
+import tech.pegasys.teku.networking.p2p.network.P2PNetwork
+import tech.pegasys.teku.networking.p2p.network.PeerAddress
+import tech.pegasys.teku.networking.p2p.network.config.GeneratingFilePrivateKeySource
+import tech.pegasys.teku.networking.p2p.peer.DisconnectReason
+import tech.pegasys.teku.networking.p2p.peer.NodeId
+import tech.pegasys.teku.networking.p2p.peer.Peer
+
+class P2PManager(
+  val privateKeyFile: Path,
+  val ipAddress: String,
+  val port: String,
+  val staticPeers: List<String>,
+) {
+  companion object {
+    private const val DEFAULT_RECONNECT_DELAY_MILLI_SECONDS = 5000L // TODO: Do we want to make this configurable?
+  }
+
+  private val log: Logger = LogManager.getLogger(this::class.java)
+  private val delayedExecutor = SafeFuture.delayedExecutor(DEFAULT_RECONNECT_DELAY_MILLI_SECONDS, TimeUnit.MILLISECONDS)
+  private val staticPeerMap = mutableMapOf<NodeId, MultiaddrPeerAddress>()
+
+  lateinit var p2pNetwork: P2PNetwork<Peer>
+
+  fun start() {
+    p2pNetwork = buildP2PNetwork()
+
+    p2pNetwork
+      .start()
+      ?.thenApply {
+        staticPeers.forEach { peer ->
+          p2pNetwork.createPeerAddress(peer)?.let { address -> addStaticPeer(address as MultiaddrPeerAddress) }
+        }
+      }?.get()
+  }
+
+  fun stop() {
+    p2pNetwork.stop()
+  }
+
+  private fun buildP2PNetwork(): P2PNetwork<Peer> {
+    val filePrivateKeySource = GeneratingFilePrivateKeySource(privateKeyFile.toString())
+    // TODO: reading/generating this key should be done early on, as it is needed for the validator as well
+    val privKeyBytes = filePrivateKeySource.privateKeyBytes
+    val privateKey = unmarshalPrivateKey(privKeyBytes.toArrayUnsafe())
+
+    return P2PNetworkFactory.build(
+      privateKey = privateKey,
+      ipAddress = ipAddress,
+      port = port,
+    )
+  }
+
+  fun addStaticPeer(peerAddress: MultiaddrPeerAddress) {
```

Comment:
Do we really need this method? I don't see how would it be useful

Reply:
That was what I was thinking.

Reply:
The will probably complain about the dead code. Since it's already written, I guess we can leave it as is, but there's a change we'll never need it